command_server/command_server.py

```diff
@@ -1,80 +1,83 @@
-        self._loop = loop or asyncio.get_event_loop()
-        self._init_server(host, port)
-
-    def start(self):
-        self._server = self._loop.run_until_complete(self._server)
-        logging.info(' %s has started.' % (self._server_type))
-
-    def stop(self):
-        self._server.close()
-        logging.info('%s has stopped.' % (self._server_type))
-
+class CommandServer(object):
 
-class StreamCommandServer(BaseCommandServer):
     _instance = None
+    _controller = None
 
-    def _init_server(self, host, port):
+    def __init__(self,  host=None, port=None, templates=None, **kwargs):
+        logging.info('Init Server on host %s:%s' % (host, port))
+        self._loop = asyncio.get_event_loop()
+        self._ws = web.WebSocketResponse()
         self._app = web.Application(loop=self._loop)
-        self._server = websockets.serve(self.process_request, host, port)
-
-    def __new__(cls, *args, **kwargs):
-        if not cls._instance:
-            cls._instance = super(StreamCommandServer, cls).__new__(cls)
-        return cls._instance
-
-    @asyncio.coroutine
-    def process_request(self, websocket, path):
-        while True:
-            yield from websocket.send(strftime("%Y-%m-%d %H:%M:%S", gmtime()))
-            yield from asyncio.sleep(2)
-        yield from websocket.close()
-
-
-class HttpCommandServer(BaseCommandServer):
-    _instance = None
+        asyncio.async(self.get_game_ctr().run())
 
-    def _init_server(self, host, port):
-        self._app = web.Application()
         self._load_routes()
         self._server = self._loop.create_server(self._app.make_handler(),
                                                 host, port)
 
-    def __init__(self, templates=None, **kwargs):
-        super().__init__(**kwargs)
         if templates:
             aiohttp_jinja2.setup(self._app,
                                  loader=jinja2.FileSystemLoader(templates))
-    
+
     def __new__(cls, *args, **kwargs):
         if not cls._instance:
-            cls._instance = super(HttpCommandServer, cls).__new__(cls)
+            cls._instance = super(CommandServer, cls).__new__(cls)
         return cls._instance
 
+    @classmethod
+    def get_game_ctr(cls):
+        if not cls._controller:
+            cls._controller = GameController(50, 50, 2)
+        return cls._controller
+
+    def start(self):
+        self._server = self._loop.run_until_complete(self._server)
+        logging.info('Server has started.')
+        self._loop.run_forever()
+
+    def stop(self):
+        self._server.close()
+        logging.info('Server has stopped.')
+
     def _load_routes(self):
         logging.debug('Loading  Application Routes:\n%s' % '\n'.join(str(r) for r in routes.ROUTES))
+        self._app.router.add_route('GET', '/ws_stream', self.ws_stream)
         for route in routes.ROUTES:
             self._app.router.add_route(*route)
 
+    @asyncio.coroutine
+    def ws_stream(self, request, *args, **kwargs):
```

Comment:
it would be nice to move some logic to WebSocketView
e.g
`elif msg.tp == aiohttp.MsgType.close:
           logging.info('websocket connection closed')
  elif msg.tp == aiohttp.MsgType.error:
        logging.info('ws connection closed with exception %s',
               self._ws.exception())`